### PR TITLE
Dev/checkout page

### PR DESCRIPTION
**Pull request by patrik-laurell**

### CODE REVIEW

File: src/dashboard/components/DonationSummary.vue [hunk 1]
```diff
@@ -127,22 +120,6 @@ export default {
...
     },
-    donateAll() {
-      this.$store.dispatch('metamask/donateAll', this.distribution).catch(e => {
-        console.error('donateAll (in vue) error:', e);
-        // We're currently not catching the emitting anywhere so we
-        // console.error for now
-        this.$emit('error', e);
-      });
-    },
-    distribute() {
```
Comment (dwilid) [inline src/dashboard/components/DonationSummary.vue]: Don't move this, it conflicts with the refactor I'm doing. My pull request (https://github.com/SuperuserLabs/thankful/pull/213) will move the distribution data to vuex store so it is accessible from anywhere.

File: src/dashboard/store/modules/metamask.ts [hunk 2]
```diff
@@ -92,20 +92,26 @@ export default {
...
-      console.log('completed donation!');
-      state.pendingDonations[donation.id].status = 'completed';
-      console.log(state.pendingDonations);
+      state.pendingDonations = {
+        ...state.pendingDonations,
+        [donation.id]: {
+          ...state.pendingDonations[donation.id],
+          status: 'completed',
+        },
+      };
```
Comment (ErikBjare) [inline src/dashboard/store/modules/metamask.ts]: I find this a bit annoying to read, wouldn't `Vue.set()` be a lot cleaner?

File: src/dashboard/pages/DashboardPage.vue [hunk 3]
```diff
@@ -127,6 +130,24 @@ export default {
...
     },
+    distribute() {
+      this.distribution = this.creatorsWithShare.map(c => {
+        return {
+          ...c,
+          funds: parseFloat((c.share * this.budget_per_month).toFixed(2)),
+        };
+      });
+    },
+    errfun(title) {
```
Comment (ErikBjare) [inline src/dashboard/pages/DashboardPage.vue]: This should probably be made a Vue global or something (Call it `$error`?).

File: src/dashboard/pages/DashboardPage.vue [hunk 4]
```diff
@@ -167,5 +188,14 @@ export default {
     });
     this.$store.dispatch('db/ensureActivities');
   },
+  async created() {
+    await this.$store.dispatch('db/ensureDonations');
+    this.distribute();
+  },
+  watch: {
+    creators() {
+      this.distribute();
```
Comment (ErikBjare) [inline src/dashboard/pages/DashboardPage.vue]: Why is this needed? Wouldn't this mean that any change to the distribution is reset if creators change somehow?

File: src/dashboard/store/modules/metamask.ts (hunk 2 above)
Comment (patrik-laurell) [reply src/dashboard/store/modules/metamask.ts]: It would. Th problem is that Vue.set didn't trigger a UI update. I think the problem is that Vue tracks state.pendingDonations, but `Vue.set(state.pendingDonations[donation.id], 'status', 'completed')` changes the pendingDonations object but does not trigger changes in the UI. Probably
```
let updatedDonation = { ...state.pendingDonation[donation.id], status: 'completed' }
Vue.set(state.pendingDonations, donation.id, updatedDonation)
```
would work though. I'll try it

File: src/dashboard/components/DonationSummary.vue (hunk 1 above)
Comment (ErikBjare) [reply src/dashboard/components/DonationSummary.vue]: While unfortunate that you both poked at the same code, the easiest would probably be to just let whoever merges last to resolve the conflicts.

File: src/dashboard/pages/CheckoutPage.vue [hunk 5]
```diff
@@ -1,18 +1,83 @@
...
+      v-list
+        v-list-tile(v-for="d in pendingDonations", :key="d.id")
+          v-list-tile-action
+            v-progress-circular(v-if="d.status === 'pending'", indeterminate, color="primary")
+            v-icon(v-else-if="d.status === 'failed'") error
+            v-icon(v-else-if="d.status === 'completed'", color="primary") check_circle
+          v-list-tile-content
+            v-list-tile-title(v-text="d.name")
+    v-stepper-content(step="3")
+      h3 All done! Now just keep surfin' baby.
```
Comment (ErikBjare) [inline src/dashboard/pages/CheckoutPage.vue]: lol, change to slightly less Patrik-y message?

Comment (patrik-laurell) [reply src/dashboard/pages/CheckoutPage.vue]: Something like 'All donations where successful! You can now go back to browsing as usual. We will remind you when it is time to donate again.' ?

File: src/dashboard/pages/DashboardPage.vue (hunk 4 above)
Comment (patrik-laurell) [reply src/dashboard/pages/DashboardPage.vue]: Yes it would. Not really needed i guess, but all of that logic will be changed by @dwilid s PR #213

File: src/dashboard/store/modules/metamask.ts (hunk 2 above)
Comment (patrik-laurell) [reply src/dashboard/store/modules/metamask.ts]: For some reason that code did not trigger the reactivity. In the console the object is changed but the UI does not update. So I will keep it like this for now

Comment (ErikBjare) [reply src/dashboard/store/modules/metamask.ts]: Fair enough

File: src/dashboard/pages/DashboardPage.vue (hunk 4 above)
Comment (ErikBjare) [reply src/dashboard/pages/DashboardPage.vue]: Alright then

File: src/dashboard/pages/CheckoutPage.vue (hunk 5 above)
Comment (ErikBjare) [reply src/dashboard/pages/CheckoutPage.vue]: That sounds reasonable
